src/oop/object_interactions/medium/q5/MainTest.java

```diff
@@ -0,0 +1,178 @@
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            String updateName = (String) test[num][2];
+            int updateEyes = (int) test[num][3];
+            Object output = testClass.callMethod(getMyNameMethodName, classInstance);
+            _assertEquals(name, output,
+                    "You have modified the provided " + getMyNameMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+            output = testClass.callMethod(getNumEyesMethodName, classInstance);
+            _assertEquals(eyes, output,
+                    "You have modified the provided " + getNumEyesMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+            output = testClass.callMethod(setMyNameMethodName, new Object[][]{{updateName, String.class}}, classInstance);
+            _assertEquals(updateName, testClass.getFieldValue(classInstance, myNameAttributeName),
+                    "You have modified the provided " + setMyNameMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+            assertNull(output,
+                    "You have modified the provided " + setMyNameMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+            output = testClass.callMethod(setNumEyesMethodName, new Object[][]{{updateEyes, int.class}}, classInstance);
+            _assertEquals(updateEyes, testClass.getFieldValue(classInstance, numEyesAttributeName),
+                    "You have modified the provided " + setNumEyesMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+            assertNull(output,
+                    "You have modified the provided " + setNumEyesMethodName + "method  on the " + className + " class. Please revert it to the original state.");
+        }
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of("Sarah", 0, "John", 1),
+                Arguments.of("Amir", 1, "Sam", 2),
+                Arguments.of("John", 0, "Jack", 0)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependent2")
+    public void correctReceiveDonationMethod(String name, int eyes, String donName, int donEyes) throws Throwable {
+        Object[][] arguments = {
+                {name, String.class},
+                {eyes, int.class}
+        };
+        Object[][] donorArguments = {
+                {donName, String.class},
+                {donEyes, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object donorInstance = testClass.createInstance(donorArguments);
+        Object[][] methodArguments = {
+                {donorInstance, Person.class}
+        };
+        Object output = null;
```

Comment:
use .hasMethod before the if statement with the .callMethod. This will get rid of the need for the output variable and the assertNull